osu.Game/Screens/SelectV2/BeatmapCarouselFilterSorting.cs

```diff
@@ -118,10 +118,10 @@ private static int compareDifficulty(BeatmapInfo a, BeatmapInfo b)
             return comparison;
         }
 
-        private static int compareUsingAggregateMax(BeatmapInfo a, BeatmapInfo b, IEnumerable<CarouselItem> items, Func<BeatmapInfo, double> func)
+        private static int compareUsingAggregateMax(BeatmapInfo a, BeatmapInfo b, Func<BeatmapInfo, double> func)
         {
-            var aMatchedBeatmaps = items.Select(i => i.Model).Cast<BeatmapInfo>().Where(beatmap => beatmap.BeatmapSet!.Equals(a.BeatmapSet));
-            var bMatchedBeatmaps = items.Select(i => i.Model).Cast<BeatmapInfo>().Where(beatmap => beatmap.BeatmapSet!.Equals(b.BeatmapSet));
+            var aMatchedBeatmaps = a.BeatmapSet!.Beatmaps.Where(bb => !bb.Hidden);
+            var bMatchedBeatmaps = b.BeatmapSet!.Beatmaps.Where(bb => !bb.Hidden);
```

Comment:
Before it's asked, yes, this may include non-matching beatmaps. I think this is fine.

Reply:
Given how things are I think there's a good chance that we're going to get asked to split out difficulties for all three sort modes touched here rather than use max across the set. So I'm fine ignoring this concern too.

Reply:
See https://github.com/ppy/osu/pull/33283 I guess. Although that PR may not work according to ~~twitch chat~~